src/jsonrpc.js

```diff
@@ -0,0 +1,87 @@
+import logger from './logger';
+
+/**
+ * This is promisify function that convert function with callback to promise
+ * @param {*} func 
+ */
+export const toPromise = func => (...args) =>
+  new Promise((resolve, reject) =>
+    func(...args, (error, result) => (error ? reject(new Error(error.message)) : resolve(result)))
+  );
+
+
+export class JsonRpc {
+  /**
+   * Initialize class local variables
+   * @param {*} addresses 
+   * @param {*} fromBlock 
+   * @param {*} toBlock 
+   * @param {*} callback 
+   */
+  constructor(addresses, fromBlock, toBlock, web3, callback = null) {
+    this.addresses = addresses.map(address => address.toLowerCase());
+    this.fromBlock = fromBlock;
+    this.toBlock = toBlock;
+    this.web3Instance = web3.eth;
+    this.callback = callback;
+  }
+
+  /**
+   * 
+   * This will return the final data structure 
+   * for the transaction resopnse
+   * @param {*} tx 
+   * @param {*} receipt  
+   * @return object
+   */
+  _getTransaction(tx, receipt) {
+    return { ...tx, ...receipt };
+  }
+
+  /**
+   * Async function that get the transaction and transaction receipt then get the 
+   * logs out of the receipt transaction then execute the callback function
+   * @param {*} txn
+   */
+  async _scanTransaction(txn) {
+    const txnReceipts = await toPromise(this.web3Instance.getTransactionReceipt)(txn.hash);
```

Comment:
It's a good idea to call `getTransactionReceipt` as it gives a chance to scan the logs before commencing if we want to store this transaction or not, However, this means we need to implement log checking in this `if` statement